### PR TITLE
SAML Authentication

### PR DESCRIPTION
**Pull request by cccs-sgaron**

### CODE REVIEW

File: assemblyline_ui/api/v4/authentication.py [hunk 1]
```diff
@@ -950,3 +1116,47 @@ def validate_otp(token, **kwargs):
...
+        "post_data": request.form.copy()
+    }
+
+
+def _make_saml_auth(request_data: Dict[str, Any] = None) -> OneLogin_Saml2_Auth:
+    request_data: Dict[str, Any] = request_data or _prepare_flask_request(request)
+    return OneLogin_Saml2_Auth(request_data, config.auth.saml.settings.as_camel_case())
+
+
+url_regex = re.compile(
```
Comment (cccs-kevin) [inline assemblyline_ui/api/v4/authentication.py:1141]: can we import this from assemblyline base?

Comment (eljeffeg) [reply assemblyline_ui/api/v4/authentication.py:1141]: I don't see why not - makes sense.

Comment (cccs-sgaron) [reply assemblyline_ui/api/v4/authentication.py:1141]: Or just use URL parse to get the host out... cause that is essentially why we need this REGEX.

Comment (cccs-sgaron) [reply assemblyline_ui/api/v4/authentication.py:1141]: Actually, this code is not needed anymore. Its not called anymore. I'll just remove it

**Pull request merged**